lisa/features/gpu.py

```diff
@@ -0,0 +1,144 @@
+
+    # download and install CUDA Driver
+    def _install_cuda_driver(self, version: str = "10.1.105-1") -> None:
+        self._log.debug("Starting CUDA driver installation")
+        cuda_repo = ""
+        os_version = self._node.os.os_version
+
+        if isinstance(self._node.os, Redhat):
+            release = os_version.release.split(".")[0]
+            cuda_repo_pkg = f"cuda-repo-rhel{release}-{version}.x86_64.rpm"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/"
+                f"compute/cuda/repos/rhel{release}/x86_64/{cuda_repo_pkg}"
+            )
+        elif isinstance(self._node.os, Ubuntu):
+            release = re.sub("[^0-9]+", "", os_version.release)
+            cuda_repo_pkg = f"cuda-repo-ubuntu{release}_{version}_amd64.deb"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/compute/"
+                f"cuda/repos/ubuntu{release}/x86_64/{cuda_repo_pkg}"
+            )
+        else:
+            raise LisaException(
+                f"Distro {self._node.os.__class__.__name__}"
+                "not supported to install CUDA driver."
+            )
+
+        # download and install the cuda driver package from the repo
+        self._node.os._install_package_from_url(f"{cuda_repo}", signed=False)
+
+    def _install_gpu_dep(self) -> None:
+        uname_tool = self._node.tools[Uname]
+        uname_ver = uname_tool.get_linux_information().uname_version
+
+        # install dependency libraries for distros
+        if isinstance(self._node.os, Redhat):
+            # install the kernel-devel and kernel-header packages
+            package_name = f"kernel-devel-{uname_ver} kernel-headers-{uname_ver}"
+            self._node.os.install_packages(package_name)
+            # mesa-libEGL install/update is require to avoid a conflict between
+            # libraries - bugzilla.redhat 1584740
+            package_name = "mesa-libGL mesa-libEGL libglvnd-devel"
+            self._node.os.install_packages(package_name)
+            # install dkms
+            package_name = "dkms"
+            self._node.os.install_packages(package_name, signed=False)
+        elif isinstance(self._node.os, Ubuntu):
+            package_name = (
+                f"build-essential libelf-dev linux-tools-{uname_ver}"
+                f" linux-cloud-tools-{uname_ver} python libglvnd-dev ubuntu-desktop"
+            )
+            self._node.os.install_packages(package_name)
+        else:
+            raise LisaException(
+                f"Distro {self._node.os.__class__.__name__}"
+                " is not supported for GPU."
+            )
+
+    def check_support(self) -> None:
+        # TODO: more supportability can be defined here
+        if not self._is_supported():
+            raise SkippedException(f"GPU is not supported with distro {self._node.os}")
+
+    def install_compute_sdk(
+        self, driver: ComputeSDK = ComputeSDK.CUDA, version: str = ""
```

Comment:
The default value of version should be `10.1.105-1` like `_install_cuda_driver`.

Reply:
we cannot have one default version since there can be different kinds of drivers. So all of them wouldn't  have one default version

Reply:
The default version is to meet for major/default test, which doesn't need to specify version. If give empty string to version, the default value actually doesn't work. Because we already give cuda to first argument, we can give cuda version together.

Reply:
Another way is to fill the default version in this method, if it's cuda, and the version is empty, fill in the `"10.1.105-1"`. If it's grid with empty version, fill in `"DEFAULT_GRID_DRIVER_URL"`.

Reply:
ok, let me fix this